### PR TITLE
Fix: Caching of rendered dbt model configs by making the dependencies attribute public

### PR DESCRIPTION
**Pull request by izeigerman**

### CODE REVIEW

File: sqlmesh/dbt/basemodel.py [hunk 1]
```diff
@@ -109,8 +119,7 @@ class BaseModelConfig(GeneralConfig):
     storage_format: t.Optional[str] = None
     path: Path = Path()
     target_schema: str = ""
-    _dependencies: Dependencies = Dependencies()
-    _variables: t.Dict[str, bool] = {}
```
Comment (izeigerman) [inline sqlmesh/dbt/basemodel.py:113]: @crericha this doesn't seem to be used anywhere so I dropped it. Let me know if this is wrong.

**Review by tobymao — APPROVED** [review]

File: sqlmesh/dbt/basemodel.py (hunk 1 above)
Comment (crericha) [reply sqlmesh/dbt/basemodel.py:113]: Yeah, it got encapsulated into dependencies.

File: sqlmesh/dbt/basemodel.py [hunk 2]
```diff
@@ -61,6 +61,16 @@ def union(self, other: Dependencies) -> Dependencies:
...
 
+    def dict(self, *args: t.Any, **kwargs: t.Any) -> t.Dict[str, t.Any]:
+        # See https://github.com/pydantic/pydantic/issues/1090
+        exclude = kwargs.pop("exclude", None) or set()
+
+        out = super().dict(*args, **kwargs, exclude={*exclude, "macros"})
+        if "macros" not in exclude:
+            out["macros"] = [macro.dict() for macro in self.macros]
+
+        return out
```
Comment (crericha) [inline sqlmesh/dbt/basemodel.py:72]: We already have an attribute_dict method. Can that be used?

Comment (izeigerman) [reply sqlmesh/dbt/basemodel.py:72]: `attribute_dict` relies on Pydantic `dict()` and fails without this change. This method overrides the Pydantic's `dict()`.

Comment (crericha) [reply sqlmesh/dbt/basemodel.py:72]: Okay, I didn't realize it was an override.

**Review by crericha — APPROVED** [review]

**Pull request merged**